### PR TITLE
Add Kakoune and Helix configuration

### LINKED ISSUE #180: Add some more IDE configurations in the user manual
In the User manual `manual.adoc`, there is a section to help everyone to set up their favorite IDE/editor. This is great.

I could probably provide a PR for at least two more text editors ([Kakoune](https://kakoune.org/) and [Helix](https://docs.helix-editor.com/)).

But first, I wanted to know if this kind of changes would be welcomed. I could understand the desire to keep the list short and only about mainstream IDE/editors. Tell me if you'd like me to contribute or else, you can just close the issue.

### PR DESCRIPTION
**Pull request by woshilapin**
Fixes #180.

I choose to put links to existing official documentation instead of copy/paste what to do with the risk of being out-of-date (especially for these young editors where things move often).

### CODE REVIEW

**Comment by lnicola** [comment]
Please add hard line breaks after each sentence, it makes reading diffs much easier.

And I'm not sure about the helix remark/emoji. I don't mind them that much, but they don't really fit the style used in the rest of the docs.

**Comment by woshilapin** [comment]
> Please add hard line breaks after each sentence, it makes reading diffs much easier.

Indeed, done.

> And I'm not sure about the helix remark/emoji. I don't mind them that much, but they don't really fit the style used in the rest of the docs.

I might have been a bit too comfortable here :sweat_smile: Let's say it was late yesterday evening! I removed it.

File: manual.adoc [hunk 1]
```diff
@@ -427,6 +427,40 @@ Then click on apply, and restart the LSP server for your rust project.
 
 https://gitlab.com/cppit/jucipp[juCi++] has built-in support for the language server protocol, and since version 1.7.0 offers installation of both Rust and rust-analyzer when opening a Rust file.
 
+=== Kakoune
+
+https://kakoune.org/[Kakoune] editor support LSP with the help of https://github.com/kak-lsp/kak-lsp[`kak-lsp`].
```
Comment (lnicola) [inline manual.adoc:432]:
```suggestion
https://kakoune.org/[Kakoune] supports LSP with the help of https://github.com/kak-lsp/kak-lsp[`kak-lsp`].
```

File: manual.adoc [hunk 2]
```diff
@@ -427,6 +427,40 @@ Then click on apply, and restart the LSP server for your rust project.
...
+    hook window -group rust-inlay-hints BufWritePost .* rust-analyzer-inlay-hints
+    hook -once -always window WinSetOption filetype=.* %{
+        remove-hooks window rust-inlay-hints
+    }
+}
+----
+
+=== Helix
+
+https://docs.helix-editor.com/[Helix] editor does support LSP by default.
```
Comment (lnicola) [inline manual.adoc:460]:
```suggestion
https://docs.helix-editor.com/[Helix] supports LSP by default.
```

**Comment by lnicola** [comment]
bors d+

changelog fix (first contribution) add install instructions for Kakoune and Helix.

File: manual.adoc (hunk 2 above)
Comment (woshilapin) [reply manual.adoc]: Fixed in ebca994.

File: manual.adoc (hunk 1 above)
Comment (woshilapin) [reply manual.adoc]: Fixed in ebca994.

**Comment by woshilapin** [comment]
First time I interact with bors. I thought I was allowed to do `bors r+` after @lnicola did `bors d+`, but seems like there is a problem. Did I do something wrong?

**Comment by lnicola** [comment]
Sorry I didn't notice it, can you port your changes to https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/user/manual.adoc?

There is a comment at the top:

```
IMPORTANT: the master copy of this document lives in the https://github.com/rust-analyzer/rust-analyzer repository
```

**Comment by woshilapin** [comment]
New PR opened in https://github.com/rust-analyzer/rust-analyzer/pull/11535.

**Pull request closed without merging**